cmd/scrape.go

```diff
@@ -0,0 +1,543 @@
+			Input         string `json:"input"`
+			To            string `json:"to"`
+			Value         string `json:"value"`
+			Author        string `json:"author"` // reward
+			RewardType    string `json:"rewardType"`
+			Address       string `json:"address"` // suicide
+			Balance       string `json:"balance"`
+			RefundAddress string `json:"refundAddress"`
+			Init          string `json:"init"` // create
+		} `json:"action,omitempty"`
+		BlockHash   string `json:"blockHash"`
+		BlockNumber int    `json:"blockNumber"`
+		Error       string `json:"error"`
+		Result      struct {
+			GasUsed string `json:"gasUsed"` // call
+			Output  string `json:"output"`
+			Address string `json:"address"` // create
+		} `json:"result"`
+		Subtraces           int           `json:"subtraces"`
+		TraceAddress        []interface{} `json:"traceAddress"`
+		TransactionHash     string        `json:"transactionHash"`
+		TransactionPosition int           `json:"transactionPosition"`
+		Type                string        `json:"type"`
+	} `json:"result"`
+	ID int `json:"id"`
+}
+
+// BlockLogs Returned value from the RPC containing all the logs for a given block.
+type BlockLogs struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  []struct {
+		Address             string   `json:"address"`
+		BlockHash           string   `json:"blockHash"`
+		BlockNumber         string   `json:"blockNumber"`
+		Data                string   `json:"data"`
+		LogIndex            string   `json:"logIndex"`
+		Removed             bool     `json:"removed"`
+		Topics              []string `json:"topics"`
+		TransactionHash     string   `json:"transactionHash"`
+		TransactionIndex    string   `json:"transactionIndex"`
+		TransactionLogIndex string   `json:"transactionLogIndex"`
+		Type                string   `json:"type"`
+	} `json:"result"`
+	ID int `json:"id"`
+}
+
+// TransReceipt - a given transaction's receipt
+type TransReceipt struct {
+	Jsonrpc string `json:"jsonrpc"`
+	Result  struct {
+		BlockHash         string        `json:"blockHash"`
+		BlockNumber       string        `json:"blockNumber"`
+		ContractAddress   string        `json:"contractAddress"`
+		CumulativeGasUsed string        `json:"cumulativeGasUsed"`
+		From              string        `json:"from"`
+		GasUsed           string        `json:"gasUsed"`
+		Logs              []interface{} `json:"logs"`
+		LogsBloom         string        `json:"logsBloom"`
+		Root              string        `json:"root"`
+		Status            interface{}   `json:"status"`
+		To                interface{}   `json:"to"`
+		TransactionHash   string        `json:"transactionHash"`
+		TransactionIndex  string        `json:"transactionIndex"`
+	} `json:"result"`
+	ID int `json:"id"`
+}
+
+// BlockInternals - carries both the traces and the logs for a block
+type BlockInternals struct {
+	Traces []byte
+	Logs   []byte
+}
+
+// getTracesFromBlock Returns all traces for a given block.
+func getTracesFromBlock(blockNum int) ([]byte, error) {
```

Comment:
I think we should change this back to Go style. So: don't nest conditionals, but instead check errors after each function call.

Reply:
Done in next push.